script.js

```diff
@@ -1,4 +1,116 @@
+var gameMode = "Lets start the game";
+var numberOfPlayers = 0;
+var numberOfDice = 0;
+var playerAnswer = [];
+var sumOfPlayerScore = [];
+var playerArray = [];
+var curPlayer = 1;
+
 var main = function (input) {
-  var myOutputValue = 'hello world';
-  return myOutputValue;
+  if (gameMode == "Lets start the game") {
+    gameMode = "Input number of players";
+    console.log(gameMode);
+    return "Welcome to Beat That! Please input number of players.";
+  }
+  if (gameMode == "Input number of players") {
+    numberOfPlayers = input;
+    if (Number.isNaN(Number(input))) {
+      return `Sorry please enter a number`;
+    } else {
+      getPlayerArray();
+      console.log(numberOfPlayers);
+      gameMode = "Input number of dice";
+      console.log(gameMode);
+      return "Please input number of dice want to play.";
+    }
+  }
+  if (gameMode == "Input number of dice") {
+    numberOfDice = input;
+    if (Number.isNaN(Number(input))) {
+      return `Sorry please enter a number`;
+    } else {
+      gameMode = "Roll the dice";
+      console.log(gameMode);
+      console.log(numberOfDice);
+      return "Press continue to roll the dice";
+    }
+  }
+  if (gameMode == "Roll the dice") {
+    var diceResult = displayRolledResult();
+    gameMode = "Key in answer";
+    console.log(gameMode);
+    return diceResult + "<br>" + "Please input your answer.";
+  }
+  if (gameMode == "Key in answer") {
+    sumOfPlayerScore[Number(curPlayer - 1)] += Number(input);
+    gameMode = "Roll the dice";
+    curPlayer += 1;
+    if (curPlayer > numberOfPlayers) {
+      gameMode = "leaderboard";
+      return `All players have input the answer. Press continue to see the result❗`;
+    }
+    return "Next Player please click continue to roll the dice";
+  }
+  if (gameMode == "leaderboard") {
+    resetGame();
+    return `${getLeaderboard()} <br> Press continue for next round❗`;
+  }
+};
+console.log(sumOfPlayerScore);
+
+var rollDice = function () {
+  var randomDecimal = Math.random() * 6;
+  var randomInteger = Math.floor(randomDecimal) + 1;
+  return randomInteger;
+};
+
+var findLargestScore = function () {
+  var j = 1;
+  var largestNumber = sumOfPlayerScore[0];
+  var largestNumberPlayer = 1;
+  while (j < sumOfPlayerScore.length) {
```

Comment:
I think this can also be a for loop like the rest of your code, it's pretty similar to the other instances of loops so abit weird if this is a while loop and the rest are for loops